stylelint-rules/no-undeclared-hds-color-tokens.js

```diff
@@ -1,33 +1,53 @@
-const fs = require('fs')
+// imports
 const stylelint = require('stylelint')
 const { report, ruleMessages, validateOptions } = stylelint.utils
 
-const tokensFileName =
-  '@hashicorp/design-system-tokens/dist/devdot/css/tokens.css'
-const tokensFile = require.resolve(tokensFileName)
-const tokensFileContent = fs.readFileSync(tokensFile).toString()
-
+// meta
 const ruleName = 'digital-plugin/no-undeclared-hds-color-tokens'
 const messages = ruleMessages(ruleName, {
-  undeclared: (undeclared) =>
-    `"${undeclared}" was not found in ${tokensFileName}`,
+  undeclared: (variableName) => {
+    return `"${variableName}" was not found in provided \`tokensSource\``
+  },
 })
 
-module.exports.ruleName = ruleName
-module.exports.messages = messages
-module.exports = stylelint.createPlugin(
+// TODO move to src/lib?
+const isBoolean = (value) => {
+  const result = typeof value === 'boolean' || value instanceof Boolean
+  return result
+}
+
+// TODO move to src/lib?
+const isString = (value) => {
+  const result = typeof value === 'string' || value instanceof String
+  return result
+}
```

Comment:
I assume we want these somewhere else (built-in Stylelint rules use these but they're not exposed), but wanted to raise it as a question first. 💭

Reply:
Ideally the code for these stylelint rules would be self contained, so if we want to abstract them out they could live in `stylelint-rules/lib`. This will make the code more portable in the future if we want to publish these as npm packages, for example!